File: algorithms-data-structures/go/src/algorithms/mixed/all-longest-strings/longest.go
package alllongeststrings

func allLongestStrings(inputArray []string) []string {
	var result []string
	longest := 0
	for _, s := range inputArray {
		if len(s) > longest {
			longest = len(s)
			result = []string{s}
		} else if len(s) == longest {
			result = append(result, s)
		}
	}
	return result
}

File: ts/src/algorithms/mixed/all-longest-strings/longest.ts
// All Longest Strings solution 1
export function allLongestStrings(inputArray: string[]): string[] {
  let currentMaxLength = inputArray[0].length;
  for (let i = 1; i < inputArray.length; i++) {
    if (currentMaxLength < inputArray[i].length) {
      currentMaxLength = inputArray[i].length;
    }
  }
  return inputArray.filter(({length}) => length === currentMaxLength);
}

// All Longest Strings solution 2
export function allLongestStringsTwo(inputArray: string[]): string[] {
  let currentMaxLength = 0;
  let result: string[] = [];
  for (let s of inputArray) {
    if (s.length > currentMaxLength) {
      currentMaxLength = s.length;
      result = [s];
    } else if (s.length === currentMaxLength) {
      result.push(s);
    }
  }
  return result;
}
